app/jobs/whats_app_adapter/three_sixty_dialog/upload_file.rb

```diff
@@ -5,43 +5,41 @@
 module WhatsAppAdapter
   module ThreeSixtyDialog
     class UploadFile < ApplicationJob
-      # rubocop:disable Metrics/AbcSize
       def perform(message_id:)
         @message_id = message_id
         message = Message.find_by(id: message_id)
-        return unless message
 
         request = message.request
         organization = Organization.find_by(id: request.organization.id)
-        return unless organization && organization.three_sixty_dialog_client_api_key.blank?
 
         request.files.each do |file|
-          base_uri = ENV.fetch('THREE_SIXTY_DIALOG_PARTNER_REST_API_ENDPOINT', 'https://stoplight.io/mocks/360dialog/360dialog-partner-api/24588693')
+          base_uri = ENV.fetch('THREE_SIXTY_DIALOG_WHATS_APP_REST_API_ENDPOINT', 'https://stoplight.io/mocks/360dialog/360dialog-partner-api/24588693')
           url = URI.parse("#{base_uri}/media")
           headers = {
-            'D360-API-KEY': organization.three_sixty_dialog_client_api_key,
-            'Content-Type': file.blob.content_type
+            'D360-API-KEY' => organization.three_sixty_dialog_client_api_key
           }
-          request = Net::HTTP::Post.new(url.to_s, headers)
-          request.body = File.read(ActiveStorage::Blob.service.path_for(file.blob.key))
+          params = {
+            'messaging_product' => 'whatsapp',
+            'file' => UploadIO.new(ActiveStorage::Blob.service.path_for(file.blob.key), file.blob.content_type)
+          }
+          request = Net::HTTP::Post::Multipart.new(url, params, headers)
```

Comment:
It looks very much different from what was there before. What is the bug in #2046 and how does the PR actually fix it? It never succeeded? Errors being swallowed? Wrong status code? What happened?

Reply:
Sorry, that is a good point, I should have written more in the description.
The reason it failed silently was that the guard clause that looked to see if there was an organization and that that organization had a 360dialog client api key was incorrect. It said to return unless the attr was blank instead of return unless it's present.

The next issue was that when I removed the Setting in favor of looking up the env variables, I copy pasted the incorrect base uri for many requests. So, I needed to update to the correct base_uri.

Then, the api changed when we migrated to the Cloud API. Now, all requests need to have the `messaging_product` param set to `whatsapp` and the upload media post request needs to be posted with form data. See https://docs.360dialog.com/partner/messaging/media-messages/upload-retrieve-delete-media#upload-media

Finally, the return structure is different so that would have been another point of failure.